src/actions/WalletActions.tsx

```diff
@@ -21,6 +26,104 @@ export interface SelectWalletTokenParams {
+): Promise<void> => {
+  if (tokenIds == null) throw new Error('Activating mainnet wallets unsupported')
+  const { account } = state.core
+  const { assetOptions } = await account.getActivationAssets({ activateWalletId: wallet.id, activateTokenIds: tokenIds })
+  const { pluginId } = wallet.currencyInfo
+  const { fiatCurrencyCode } = wallet
+
+  // See if there is only one wallet option for activation
+  if (assetOptions.length === 1 && assetOptions[0].paymentWalletId != null) {
+    const { paymentWalletId, tokenId } = assetOptions[0]
+    const activationQuote = await account.activateWallet({
+      activateWalletId: wallet.id,
+      activateTokenIds: tokenIds,
+      paymentWalletId,
+      paymentTokenId: tokenId
+    })
+    const tokensText = tokenIds.map(tokenId => {
+      const { currencyCode, displayName } = getToken(wallet, tokenId) ?? {}
+      return `${displayName} (${currencyCode})`
+    })
+    const tileTitle = tokenIds.length > 1 ? s.strings.activate_wallet_tokens_scene_tile_title : s.strings.activate_wallet_token_scene_tile_title
+    const tileBody = tokensText.join(', ')
+
+    const { networkFee } = activationQuote
+    const { nativeAmount: nativeFee, currencyPluginId, tokenId: feeTokenId } = networkFee
+    if (currencyPluginId !== pluginId) throw new Error('Internal Error: Fee asset mismatch.')
+
+    const paymentCurrencyCode = getCurrencyCode(wallet, feeTokenId)
+
+    const exchangeNetworkFee = await wallet.nativeToDenomination(nativeFee, paymentCurrencyCode)
+    const feeDenom = getDisplayDenomination(state, pluginId, paymentCurrencyCode)
+    const displayFee = div(nativeFee, feeDenom.multiplier, log10(feeDenom.multiplier))
+    let fiatFee = convertCurrencyFromExchangeRates(state.exchangeRates, paymentCurrencyCode, fiatCurrencyCode, exchangeNetworkFee)
+    if (lt(fiatFee, '0.001')) fiatFee = '<0.001'
+    fiatFee = round(fiatFee, -3)
+    const feeString = `${displayFee} ${feeDenom.name} (${fiatFee} ${fiatCurrencyCode.replace('iso:', '')})`
+
+    navigation.navigate('confirmScene', {
+      titleText: s.strings.activate_wallet_token_scene_title,
+      bodyText: s.strings.activate_wallet_token_scene_body,
+      infoTiles: [
+        { label: tileTitle, value: tileBody },
+        { label: s.strings.mining_fee, value: feeString }
+      ],
+      onConfirm: (resetSlider: () => void) => {
+        if (lt(wallet.balances[paymentCurrencyCode] ?? '0', nativeFee)) {
+          const msg = tokenIds.length > 1 ? s.strings.activate_wallet_tokens_insufficient_funds_s : s.strings.activate_wallet_token_insufficient_funds_s
+          Airship.show<'ok' | undefined>(bridge => (
+            <ButtonsModal
+              bridge={bridge}
+              title={s.strings.create_wallet_account_unfinished_activation_title}
+              message={sprintf(msg, feeString)}
+              buttons={{ ok: { label: s.strings.string_ok } }}
+            />
+          ))
+          navigation.pop()
+          return
+        }
```

Comment:
If we can detect the error here, why not do it earlier and not even show the scene in the first place?

Reply:
I didn't want to recreate all the content shown to the user about wallet activation. By showing the scene first, they can see all the content and the cost to activate.

Reply:
Makes sense.